plat/Config.uk

```diff
@@ -12,6 +12,15 @@ config EARLY_PRINT_PL011_UART_ADDR
 	help
 		Pl011 serial address used by early debug console.
 
+config UKPLAT_LCPU_MAXCOUNT
+        int "Maximum number of supported logical CPUs"
+        range 1 256
+        default 1
+
+config HAVE_SMP
```

Comment:
Is this option intentionally invisible?

Reply:
Yes, the option is enabled automatically if you set LCPU_MAXCOUNT to a value > 1. Then it is in line with the other `HAVE_` options but without having to activate SMP first and then select the number of supported CPUs.

Reply:
Ah okay. The truth is that I find it a bit against my intuition :) If I was trying to enable SMP I would look for an explicit option similarly to Linux (Processor type and features -> Symmetric multi-processing support). Btw, would it make more sense to have `LCPU_MAXOUNT` as a runtime parameter? That would allow one to test and release a single image that by default starts all cores, but also allows applying a limit on a per-deployment basis.

Reply:
ok, I see. This comes from the fact that in UK the `HAVE_` variables have always been hidden in kconfig (see HAVE_LIBC etc.). So when Razvan asked how to do it, Simon came up with this solution and I liked it.

Having `LCPU_MAXCOUNT` as runtime parameter is not possible because it is used to statically scale the CPUs array in the SMP implementations. Of course you can add a runtime parameter that restricts the number of CPUs actually initialized, thus the name `MAXCOUNT` (i.e., you are not forced to start all). Btw: For Unikraft, we generally should prefer compile-time constants vs. runtime parameters where possible, so compile often and specialize as much as possible.

Reply:
Hmmm okay. On the one hand I understand the motivation. One the other hand, this has quite some implications in corporate environments, as the bigger the organization the higher the cost a release comes with. As such, people don't want to have separate releases for different configuration variants. Anyway, as long as everyone is aware of the potential implications it's just one choice over another. For `LCPU_MAXCOUNT` in particularly it's not really important.

Having said the above,
```
Reviewed-by: Michalis Pappas <michalis.pappas@opensynergy.com>
```

Reply:
Hi @michpappas,

we want to have this in for the 0.6 release next week. Everything is set up, so you can now approve the PR and add your Reviewed-by tag 😃 Thanks!

Reply:
Hey @marcrittinghaus, done thanks a lot!